cmd/server-manager/plugins/events.lua

```diff
@@ -11,21 +15,50 @@ function onEventStart(encodedRaceConfig, encodedServerOpts, encodedEntryList)
     local raceConfig = json.decode(encodedRaceConfig)
     local serverOpts = json.decode(encodedServerOpts)
     local entryList = json.decode(encodedEntryList)
+    
+	--Getting location from track_ui.json
+	location = getTrackInfo(raceConfig, serverOpts)
+	
+	-- if you want to manually set the location for tracks without location info uncomment this line and set the location, you can download a city list here: http://bulk.openweathermap.org/sample/
+	-- location = "Manchester,uk"
+	
+	if location == nil then
+		--Set location to Manchester UK if no trackinfo is found
+		location = "Manchester,UK"
+	end
 
     -- Uncomment these lines and run the function (start any event) to print out the structure of each object.
     --print("Race Config:", utils.dump(raceConfig))
     --print("Server Options:", utils.dump(serverOpts))
     --print("Entry List:", utils.dump(entryList))
-
     -- Function block NOTE: this hook BLOCKS, make sure your functions don't loop forever!
-    -- in order to use the weatherAPI you need to get a free API key from https://openweathermap.org/
-    -- uncomment functions to enable them!
-    --raceConfig, serverOpts = weatherAPI(raceConfig, serverOpts, "get-an-api-key-from-https://openweathermap.org/")
+	--Uncomment this line to set Weather API On, don't forget to put you openweathermap API key line 55
+	raceConfig, serverOpts = weatherAPI(raceConfig, serverOpts, "get-an-api-key-from-https://openweathermap.org/")
 
-    -- Encode block, you probably shouldn't touch these either!
+-- Encode block, you probably shouldn't touch these either!
     return json.encode(entryList), json.encode(serverOpts), json.encode(raceConfig)
 end
 
+-- get track location from his ui.json file, set dynamic weather on if a location is found
+-- if not, use weather in custom race config
+function getTrackInfo(raceConfig, serverOpts)
+    local track = raceConfig["Track"]
+    local layout = raceConfig["TrackLayout"]
+    local trackPath = contentPath .. "/tracks/" .. track .. "/ui/" .. layout
+    local encodedTrackJson = utils.jsonOpen(trackPath, "ui_track.json")
+    local trackJson = json.decode(encodedTrackJson)
+    countryFull = trackJson["country"]
+    city = trackJson["city"]
+    local encodedCountryCodes = utils.jsonOpen(jsonPath, "countryCodes.json")
+    local countryCodes = json.decode(encodedCountryCodes)
+    location = city .. "," .. countryCodes[countryFull]
```

Comment:
Can't city and countryCodes be nil here? There was a check for it before, not sure why it's gone but I don't think it should be. The negative flow here would panic on this line.

Best to check them and return nil if they are nil.